spec/numberformat.html

```diff
@@ -814,9 +814,9 @@ <h1>GetNumberFormatPattern ( _numberFormat_, _x_ )</h1>
             1. Let _pattern_ be _patterns_.[[negativePattern]].
         1. Else,
           1. Assert: _signDisplay_ is *"exceptZero"*.
-          1. If _x_ is 0 or _x_ is -0 or _x_ is *NaN*, then
+          1. If _x_ is 0<sub>𝔽</sub> or _x_ is -0<sub>𝔽</sub> or _x_ is *NaN*, then
```

Comment:
This operation is also generic over both Number and BigInt, so

```suggestion
          1. If _x_ is *NaN* or ℝ(_x_) is 0, then
```

(ℝ requires its argument to be finite, so we have to rule out the infinities and NaN before calling it. I'm pretty sure the infinities are screened out at the callsite, so I didn't guard against them here, but if I'm wrong this should be `or _x_ is finite and ℝ(_x_) is 0`.)

Reply:
Thanks for calling attention to the infinities in particular. I don't think those are screened at the call site: [PartitionNumberPattern](https://tc39.es/ecma402/#sec-partitionnumberpattern) has a branch for them, but it still ends up passing them to this abstract operation.

Is there any ambiguity in the prose, "If A or B and C", though?

Reply:
I suppose it's somewhat ambiguous, though I think in this context there's no other coherent reading.

But it's probably best to be more explicit. You can distinguish the cases as in `If A, or if B and C`, maybe.

Reply:
Sounds good to me!